### PR TITLE
add pre-processor to basic decoder

### PR DESCRIPTION
**Pull request by Undermaken**
this PR aims to add a processor function to basic decoder.
In this way the response can be altered before the decoder gives it back.

### CODE REVIEW

File: src/requests.ts [hunk 1]
```diff
@@ -540,20 +540,27 @@ export const ApiHeaderJson: RequestHeaderProducer<{}, "Content-Type"> = () => ({
...
   S extends number,
   R,
   O = R,
   H extends string = never
->(status: S, type: t.Type<R, O>): ResponseDecoder<IResponseType<S, R, H>> {
+>(
+  status: S,
+  type: t.Type<R, O>,
+  // tslint:disable-next-line: no-any
+  processor: (i: any) => any = _ => _
```
Comment (cloudify) [inline src/requests.ts]: perhaps `preprocessor` is more suitable as it preprocesses the json before decoding it

```suggestion
  preprocessor: (i: any) => any = _ => _
```

File: src/__tests__/requests.test.ts [hunk 2]
```diff
@@ -39,6 +39,18 @@ const getSimpleT: GetSimpleT = {
   url: params => `/api/v1/simples/${params.id}`
 };
 
+const getSimpleTP: GetSimpleT = {
+  headers: () => ({
+    Authorization: "Bearer: 123"
+  }),
+  method: "get",
+  query: params => ({ param1: `${params.p1}`, param2: params.p2 }),
+  response_decoder: basicResponseDecoder(SimpleModel, _ => {
```
Comment (cloudify) [inline src/__tests__/requests.test.ts]: no need for curly braces

```suggestion
  response_decoder: basicResponseDecoder(SimpleModel, _ => processedValue)
```

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Please include a Pivotal story at the beginning of the PR title (see below).</td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6a1bee3c157c5ba9795cf052e52ffdc5c3c0234a
</p>

**Comment by cloudify** [comment]
Released in v5.1.0

**Pull request merged**